parts/k8s/cloud-init/artifacts/cse_main.sh

```diff
@@ -123,7 +124,7 @@ if [ -f $CUSTOM_SEARCH_DOMAIN_SCRIPT ]; then
 fi
 
 if [[ "$CONTAINER_RUNTIME" == "docker" ]]; then
-    ensureDocker
+    ensureDocker || exit $ERR_DOCKER_START_FAIL
```

Comment:
why this change?

Reply:
This doesn't match the pattern in the rest CSE which is to have individual exit codes per line instead of returning non-zero and then having a broad exit code for the overall function.

Reply:
looking at the `ensureDocker` func this would happen if the json file isn't completely written in time so `ERR_FILE_WATCH_TIMEOUT` would be a more accurate exit code for that specific error. Let's replace the return by an exit in the func.

Reply:
I pushed a change that actually respects the return from "systemd start docker". I think pairing that with the (pre-existing and unused) `$ERR_DOCKER_START_FAIL` failure exit code makes sense.

Reply:
Moved this inside the ensure func to be more consistent, but for the record I think pushing those error responses into the funcs themselves is less expressive/maintainable (possible improvements to consider for another day)